### PR TITLE
Push permission

### PR DESCRIPTION
**Pull request by kosssi**
On a 3 façons différentes pour activer les push:
- [lors de l'initialisation](https://github.com/cozy/cozy-banks/pull/288/files#diff-05c90ee3c3a1a8d4973ada444d68c5c1R60)
- [lors d'un changement sur l'interface web et que Pouch rapatrie le changement](https://github.com/cozy/cozy-banks/pull/288/files#diff-bbc287eafd55b40c9b3d44b08d20efdaR52)
- [lors d'un changement dans les settings](https://github.com/cozy/cozy-banks/pull/288/files#diff-6b1a58c949aa39b33365af09e5938e1aR48)

### CODE REVIEW

File: src/ducks/categories/CategoriesPage.jsx [hunk 1]
```diff
@@ -36,7 +35,7 @@ class CategoriesPage extends Component {
 
   onWithIncomeToggle = checked => {
     const { settingsCollection, dispatch } = this.props
-    const settings = getSettings(settingsCollection)
+    const settings = get(settingsCollection, 'data[0]')
```
Comment (drazik) [inline src/ducks/categories/CategoriesPage.jsx]: Why don't you use `getSettingsFromState` in this component, like you did in `Balance` ?

File: src/ducks/settings/Notifications.jsx [hunk 2]
```diff
@@ -44,11 +45,12 @@ class Notifications extends Component {
 
     settings.notifications[setting].enabled = checked
     dispatch(updateOrCreate(settings))
+    dispatch(registerPushNotifications())
   }
 
   onChangeValue = (setting, value) => {
     const { settingsCollection, dispatch } = this.props
-    let settings = getSettings(settingsCollection)
+    let settings = get(settingsCollection, 'data[0]')
```
Comment (drazik) [inline src/ducks/settings/Notifications.jsx]: Same question about `getSettingsFromState` here

File: src/ducks/categories/CategoriesPage.jsx [hunk 3]
```diff
@@ -36,7 +36,7 @@ class CategoriesPage extends Component {
 
   onWithIncomeToggle = checked => {
     const { settingsCollection, dispatch } = this.props
-    const settings = getSettings(settingsCollection)
+    const settings = getSettingsFromCollection(settingsCollection)
```
Comment (drazik) [inline src/ducks/categories/CategoriesPage.jsx]: Why don't you use `getSettingsFromState` like in `Balance` here ?

File: src/ducks/settings/Notifications.jsx [hunk 4]
```diff
@@ -35,7 +36,7 @@ class Notifications extends Component {
 
   onToggle = (setting, checked) => {
     const { settingsCollection, dispatch } = this.props
-    let settings = getSettings(settingsCollection)
+    let settings = getSettingsFromCollection(settingsCollection)
```
Comment (drazik) [inline src/ducks/settings/Notifications.jsx]: Why don't you use `getSettingsFromState` like in `Balance` here ?

File: src/ducks/mobile/push.js [hunk 5]
```diff
@@ -1,10 +1,20 @@
...
+  const state = getState()
+
+  const settings = getSettingsFromState(state)
+  const client = state.mobile.client
+
+  return startPushNotifications(settings, client)
+}
+
+export const startPushNotifications = (settings, client) => {
+  if (push || !isNotificationEnabled(settings)) {
```
Comment (ptbrowne) [inline src/ducks/mobile/push.js]: I would put the `isNotificationEnabled` outside of this function. This way, this function does not need the `settings`.

**Pull request merged**